src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -701,6 +701,11 @@ pass::EliminateTranspose::EliminateTranspose() {
         }
 
         const auto& order_values = order_const->cast_vector<int64_t>();
+        // Don't eliminate Transpose when 0 size of order_value, which will make single TransPose ops to be eliminated
+        // when preprocess is applied by default in case of legacy ov api + plugin api 2.0. In which Convert ops will be
+        // added by preprocess but Transpose will be removed.
+        if (order_values.size() == 0)
+            return false;
```

Comment:
Not quite clear how it is possible to get empty `input_order` value. Is it for a scalar?

Reply:
Can you please elaborate more what is the pre-processing applied (mean/scale and input)?

Reply:
We can see test case, for example smoke_Transpose4D tests:
https://github.com/openvinotoolkit/openvino/blob/5574acc6ca0d5a32bfdf4214bb4b891d901041cf/src/plugins/intel_cpu/tests/functional/shared_tests_instances/single_layer_tests/transpose.cpp#L39-L45

There is `{}` in `order4D`, which will cause `order_values.size() == 0` as above.

While for Legacy api + plugin api 2.0, we can see preprocess is added by default:
https://github.com/openvinotoolkit/openvino/blob/5574acc6ca0d5a32bfdf4214bb4b891d901041cf/src/inference/src/dev/core_impl.cpp#L600-L608

It will cause new ops are added into graph, it will cause Transpose single layer tests become non-single layer and Transpose will be eliminated.